File: app/src/main/java/com/jaykallen/searchapi/model/Teams.java
package com.jaykallen.searchapi.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Teams {
    @SerializedName("teams")
    @Expose
    private List<Team> teams = null;

    public List<Team> getTeams() {
        return teams;
    }

    public void setTeams(List<Team> teams) {
        this.teams = teams;

    }
}

File: app/src/main/java/com/jaykallen/searchapi/model/Teams.kt
package com.jaykallen.searchapi.model

import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Teams (
    @SerializedName("teams") var teams: List<Team>? = null
) : Parcelable